overlord/devicestate/handlers_gadget.go

```diff
@@ -201,7 +201,13 @@ func (m *DeviceManager) doUpdateGadgetAssets(t *state.Task, _ *tomb.Tomb) error
 		// attempt to modify modeenv inside, which implicitly is
 		// guarded by the state lock; on top of that we do not expect
 		// the update to be moving large amounts of data
-		return gadgetUpdate(model, *currentData, *updateData, snapRollbackDir, updatePolicy, updateObserver)
+		if err := gadgetUpdate(model, *currentData, *updateData, snapRollbackDir, updatePolicy, updateObserver); err != nil {
+			return err
+		}
+		if updateObserver == nil {
+			return nil
+		}
+		return observeTrustedBootAssets.UpdateBootEntry()
```

Comment:
Looks like this is the handler for when the gadget updates and EFI boot variables need to be reset, yes?

Reply:
Correct.

Reply:
doesn't seem to be a unit test for this change?

Reply:
It seems there is no unit tests at all for `doUpdateGadgetAssets`. So I have to try to add that in the first place.